crates/weaver_semconv/src/semconv.rs

```diff
@@ -51,7 +123,16 @@ pub struct SemConvSpecWithProvenance {
     pub provenance: Provenance,
 }
 
-impl SemConvSpec {
+/// A wrapper for a [`SemConvSpec`] with its provenance.
+#[derive(Debug, Clone)]
+pub struct SemConvSpecV1WithProvenance {
```

Comment:
Couldn’t we make `SemConvSpecWithProvenance` generic to accept both v1 and v2 semconv specs? I haven't looked in detail to see if it’s possible. It's more of a comment for consideration.